src/include/uapi/ipc/topology.h

```diff
@@ -197,6 +197,16 @@ struct sof_ipc_comp_tone {
 	int32_t ramp_step;
 } __attribute__((packed));
 
+/** \brief Types of processing components */
+enum sof_ipc_process_type {
```

Comment:
I know that you want to just align headers, but why do we add this when we have comp_type already. Looks like a possible redundancy for me. Also @plbossart was right that FW code doesn't use this type at all, maybe because we didn't see need for that for the reason I wrote above.

Reply:
Yep, it feels like we need an ABI change to be future-proof with the component_type (high-level  category having impact on firmware behavior like hw_params handling, etc) and subtype (which should really not be an enum if we want to scale but rather a UUID).

Reply:
So generally it doesn' t make any sense to have "mirrored" files if they have different items, which are not needed by both sides. Absolutely impossible to maintain. So actually why are we even parsing this in the kernel? This is like topology/fw internal ids...

Reply:
@juimonen @plbossart Are you going to make some other changes? You can keep these additional types in kernel, but I'd like to not have redundant definitions in shared API when they are not used by FW.

Reply:
we can drop this for now, we'll have to redo the component management anyways, the current direction with enums is not scalable.